Explorer/Assets/DCL/InWorldCamera/ScreencaptureCamera/Systems/CaptureScreenshotSystem.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Cysharp.Threading.Tasks;
+using DCL.Character.Components;
+using DCL.CharacterCamera;
+using DCL.Diagnostics;
+using DCL.InWorldCamera.CameraReelStorageService;
+using DCL.InWorldCamera.ScreencaptureCamera.UI;
+using DCL.Multiplayer.Profiles.Entities;
+using DCL.Profiles;
+using ECS.Abstract;
+using System.Threading;
+using UnityEngine;
+using Utility;
+
+namespace DCL.InWorldCamera.ScreencaptureCamera.Systems
+{
+    [UpdateInGroup(typeof(CameraGroup))]
+    [UpdateAfter(typeof(ToggleInWorldCameraActivitySystem))]
+    [LogCategory(ReportCategory.IN_WORLD_CAMERA)]
+    public partial class CaptureScreenshotSystem : BaseUnityLoopSystem
+    {
+        private readonly ScreenRecorder recorder;
+        private readonly ScreenshotMetadataBuilder metadataBuilder;
+        private readonly ScreenshotHudView hud;
+        private readonly DCLInput.InWorldCameraActions inputSchema;
+
+        private readonly ICoroutineRunner coroutineRunner;
+        private readonly ICameraReelStorageService cameraReelStorageService;
+        private readonly CancellationTokenSource ctx;
+        private readonly Entity playerEntity;
+
+        private SingleInstanceEntity cameraEntity;
+
+        public CaptureScreenshotSystem(World world, ScreenRecorder recorder, DCLInput.InWorldCameraActions inputSchema,
+            ScreenshotHudView hud, Entity playerEntity, ScreenshotMetadataBuilder metadataBuilder, ICoroutineRunner coroutineRunner,
+            ICameraReelStorageService cameraReelStorageService)
+            : base(world)
+        {
+            this.recorder = recorder;
+            this.inputSchema = inputSchema;
+            this.hud = hud;
+            this.playerEntity = playerEntity;
+            this.metadataBuilder = metadataBuilder;
+            this.coroutineRunner = coroutineRunner;
+            this.cameraReelStorageService = cameraReelStorageService;
+
+            ctx = new CancellationTokenSource();
```

Comment:
Could use a better name than ctx.

Reply:
`ctx` and `ct` is common here in the team. I will keep it

Reply:
What does "x" stand for?